### ISSUE TITLE
Right click issue on fancy menu items and left click in mainmenu items

### ISSUE THREAD

**Issue by stefonarch** [lxqt/lxqt-panel#2120]
##### Steps to Reproduce (for bugs)

1.
2.
3.
4.

##### System Information

* LXQt Version:
* Distribution & Version:
* Qt Version:
* liblxqt Version:

**Comment by tsujan** [comment]
> The panel restarts in both conditions on niri but I don't find a backtrace file.

If there's no coredump file, then the panel's process is stopped somehow, without any crash.

> which makes me think of an issue in the panel itself...

On the contrary, there's nothing about lxqt-panel in it.

**Comment by stefonarch** [comment]
Searching for the error the first link:
https://bugreports.qt.io/browse/QTBUG-67988
> but widgets sometimes sets the parent as the toplevel instead of the menu you would think of as the parent

**Comment by tsujan** [comment]
> Searching for the error the first link:

And it shows that the problem is not in lxqt-panel. Since it happens only with niri, niri should be the culprit.

**Comment by stefonarch** [comment]
Not niri itself but smithay then.

**Comment by tsujan** [comment]
> Searching for the error the first link:
> https://bugreports.qt.io/browse/QTBUG-67988

Remove the menu-bar of FeatherPad or PCManFM-Qt or …, run it from a terminal, click its menu toolbar button to show its menu, open a sub-menu, and see the message in the terminal.

The last sentence is "Please fix the transient parent of the popup." It's nonsense; No one with an experience in Qt coding could say it.

**Comment by tsujan** [comment]
Isn't this fixed in niri?

**Comment by stefonarch** [comment]
Should open an issue in smithay about it probably.

**Comment by stefonarch** [comment]
Investigating we have more issues here:
1. bookmarks dialog in fancy menu
2. directorymenu's submenus
3. Pause idleness checks from icon in tray
4. No submenus opening on the deskttrop window

1-3 restart the panel.

**Comment by tsujan** [comment]
> Investigating we have more issues here:

*We* don't; niri does. Just a small correction ;)

**Comment by tsujan** [comment]
What about removing the niri support until these issues are fixed? They're too serious, IMO.

**Comment by stefonarch** [comment]
I don't think we should remove it, it's all  labeled experimental and 98% of our features are working. 

We have also https://github.com/hyprwm/Hyprland/issues/7737#issuecomment-2466808116 in Hyprland and https://github.com/swaywm/sway/issues/8100 in sway.

Adding them to the table in the wiki if there is space left would be enough.

**Comment by stefonarch** [comment]
Done https://github.com/lxqt/lxqt/wiki/ConfigWaylandSettings

**Comment by cmeissl** [comment]
should be fixed once https://github.com/Smithay/smithay/pull/1591 is merged and niri is updated to this commit
